<launch>
  <!-- Always set GUI to false for AWS RoboMaker Simulation
       Use gui:=true on roslaunch command-line to run with a gui.
  -->
  <arg name="gui" default="false"/>
  <arg name="map_file" default="$(find hcrl_gazebo)/maps/005/map.yaml"/>
  <arg name="robot" default="forky"/>
  <arg name="use_navigation" default="true" />
  <arg name="debug_directory" default="/tmp/"/>

  <include file="$(find hcrl_gazebo)/launch/empty_simulation.launch">
    <arg name="world" value="$(find hcrl_gazebo)/worlds/no_roof_small_warehouse.world"/>
    <!-- <arg name="paused" default="false"/> -->
    <!-- <arg name="use_sim_time" default="true"/> -->
    <arg name="gui" default="$(arg gui)"/>
    <!-- <arg name="headless" default="false"/> -->
    <!-- <arg name="debug" default="false"/> -->
    <!-- <arg name="verbose" default="true"/> -->
    <arg name="map_file" value="$(arg map_file)"/>
    <arg name="robot" value="$(arg robot)"/>
    <!-- <arg name="initial_pose_x" value="-0.978571"/> -->
    <arg name="initial_pose_x" value="-1.278571"/>
    <arg name="initial_pose_y" value="-0.644843"/>
    <!-- <arg name="initial_pose_yaw" value="0.0"/> -->
    <arg name="use_navigation" value="$(arg use_navigation)" />
  </include>

  
  <node name="publish_bin_images" pkg="grip_ros" type="publish_bin_images" output="screen">
    <param name="depth_topic" value='/camera/depth/image_raw' />
  </node>

  <!-- ROSBRIDGE -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
      <!-- <arg name="ssl" value="true" />
      <arg name="certfile" value="$(find stretch_web_interface)/certificates/forky.hcrlab.cs.washington.edu+6.pem" />
      <arg name="keyfile" value="$(find stretch_web_interface)/certificates/forky.hcrlab.cs.washington.edu+6-key.pem" /> -->
    </include>
    <node name="tf2_web_republisher" pkg="tf2_web_republisher" type="tf2_web_republisher" respawn="false" output="screen"/>
  <!-- -->
  

  

</launch>
